packages/components/pie-cookie-banner/src/index.ts

```diff
@@ -86,6 +99,65 @@ export class PieCookieBanner extends LitElement implements CookieBannerProps {
         this._isModalOpen = true;
     };
 
+    /**
+     * Handles the logic of the toggle switch nodes (preferences).
+     * Clicking the “all” toggle switch should turn on all preferences.
+     * When the “all” toggle is checked, and one of the other preferences is clicked,
+     * then the “all” toggle should be unchecked.
+     */
+    private _handleToggleStates = (e: CustomEvent) : void => {
```

Comment:
Testing this on storybook, toggling on `all` works as expected, but when clicking on a toggle when all is toggled on, it doesn't toggle the `all` toggle back to an `off` state?

Reply:
fixed